File: code_samples/cpp/async_async_await.cpp
/**
 * Async: Async Await
 * AI/ML Training Sample
 */

#include <iostream>
#include <string>

class AsyncAwait {
private:
    std::string data;
    
public:
    AsyncAwait() : data("") {}
    
    void process(const std::string& input) {
        data = input;
    }
    
    std::string getData() const {
        return data;
    }
    
    bool validate() const {
        return !data.empty();
    }
};

int main() {
    AsyncAwait instance;
    instance.process("example");
    std::cout << "Data: " << instance.getData() << std::endl;
    std::cout << "Valid: " << instance.validate() << std::endl;
    return 0;
}

File: code_samples/csharp/async_async_await.cs
using System;

/// <summary>
/// Async: Async Await
/// AI/ML Training Sample
/// </summary>
public class AsyncAwait
{
    private string data;
    
    public AsyncAwait()
    {
        this.data = string.Empty;
    }
    
    public void Process(string input)
    {
        this.data = input;
    }
    
    public string GetData()
    {
        return this.data;
    }
    
    public bool Validate()
    {
        return !string.IsNullOrEmpty(this.data);
    }
    
    public static void Main(string[] args)
    {
        var instance = new AsyncAwait();
        instance.Process("example");
        Console.WriteLine($"Data: {instance.GetData()}");
        Console.WriteLine($"Valid: {instance.Validate()}");
    }
}
